cardano_node_tests/tests/test_plutus_spend_build.py

```diff
@@ -1209,3 +1239,69 @@ def test_collateral_is_txin(
 
         dbsync_utils.check_tx(cluster_obj=cluster, tx_raw_output=tx_output_step1)
         dbsync_utils.check_tx(cluster_obj=cluster, tx_raw_output=tx_output_step2)
+
+
+@pytest.mark.testnets
+class TestNegative:
+    """Tests for txin locking using Plutus smart contracts that are expected to fail."""
+
+    @allure.link(helpers.get_vcs_link())
+    @pytest.mark.testnets
+    @hypothesis.given(redeemer_number=st.integers(min_value=-2 ^ 64, max_value=2 ^ 64))
+    @common.hypothesis_settings()
+    def test_guessing_game_pbt(
+        self,
+        cluster_lock_guessing_game: clusterlib.ClusterLib,
+        payment_addrs_lock_guessing_game: List[clusterlib.AddressRecord],
+        fund_script_guessing_game: clusterlib.TxRawOutput,
+        redeemer_number: int,
+    ):
+        """Try to spend a locked UtxO with and an invalid redeemer.
+
+        Expect failure.
+        """
+        hypothesis.assume(redeemer_number != 42)
+
+        cluster = cluster_lock_guessing_game
+        temp_template = common.get_test_id(cluster)
+
+        if redeemer_number % 3 == 0 or redeemer_number % 2 == 0:
+            redeemer_file = "redeemer_file.datum"
+
+            with open(redeemer_file, "w", encoding="utf-8") as outfile:
+                if redeemer_number % 3 == 0:
+                    json.dump({"constructor": 0, "fields": [{"int": redeemer_number}]}, outfile)
+                else:
+                    json.dump({"int": redeemer_number}, outfile)
+
+            plutus_op = plutus_common.PlutusOp(
+                script_file=plutus_common.GUESSING_GAME_PLUTUS,
+                datum_file=plutus_common.DATUM_42_TYPED,
+                redeemer_file=Path(redeemer_file),
+            )
+        else:
+            redeemer_value = str(redeemer_number)
+
+            plutus_op = plutus_common.PlutusOp(
+                script_file=plutus_common.GUESSING_GAME_PLUTUS,
+                datum_file=plutus_common.DATUM_42_TYPED,
+                redeemer_value=redeemer_value,
+            )
```

Comment:
This can be flatten / shorten / simplified to e.g.:
```python
        redeemer_file = "redeemer_file.datum"

        redeemer_content = {}
        if redeemer_number % 3 == 0:
            redeemer_content = {"constructor": 0, "fields": [{"int": redeemer_number}]}
        elif redeemer_number % 2 == 0:
            redeemer_content = {"int": redeemer_number}

        if redeemer_content:
            with open(redeemer_file, "w", encoding="utf-8") as outfile:
                json.dump(redeemer_content, outfile)

        plutus_op = plutus_common.PlutusOp(
            script_file=plutus_common.GUESSING_GAME_PLUTUS,
            datum_file=plutus_common.DATUM_42_TYPED,
            redeemer_file=Path(redeemer_file) if redeemer_content else None,
            redeemer_value=None if redeemer_content else str(redeemer_number),
        )
```